pyaedt/modeler/cad/component_array.py

```diff
@@ -0,0 +1,843 @@
+
+    @property
+    def b_vector_choices(self):
+        """B vector name choices.
+
+        Returns
+        -------
+        list
+           Lattice vector names.
+        """
+        return list(self._app.get_oo_property_value(self._omodel, self.name, "B Vector/Choices"))
+
+    @property
+    def a_vector_name(self):
+        """A vector name.
+
+        Returns
+        -------
+        str
+           Lattice vector name.
+        """
+        return self._app.get_oo_property_value(self._omodel, self.name, "A Vector")
+
+    @a_vector_name.setter
+    def a_vector_name(self, val):
+        if val in self.a_vector_choices:
+            self._oarray.SetPropValue("A Vector", val)
+        else:
+            self._logger.warning("A vector name not available")
+
+    @property
+    def b_vector_name(self):
+        """B vector name.
+
+        Returns
+        -------
+        str
+           Lattice vector name.
+        """
+        return self._oarray.GetPropValue("B Vector")
+
+    @b_vector_name.setter
+    def b_vector_name(self, val):
+        if val in self.b_vector_choices:
+            self._oarray.SetPropValue("B Vector", val)
+        else:
+            self._logger.warning("B vector name not available")
+
+    @property
+    def a_size(self):
+        """A cell count.
+
+        Returns
+        -------
+        int
+           Number of cells in A direction.
+        """
+        return int(self._app.get_oo_property_value(self._omodel, self.name, "A Cell Count"))
+
+    @a_size.setter
+    def a_size(self, val):  # pragma: no cover
+        # Bug in 2024.1, not possible to change cell count.
+        # self._oarray.SetPropValue("A Cell Count", val)
+        pass
+
+    @property
+    def b_size(self):
+        """B cell count.
+
+        Returns
+        -------
+        int
+           Number of cells in B direction.
+        """
+        return int(self._app.get_oo_property_value(self._omodel, self.name, "B Cell Count"))
+
+    @b_size.setter
+    def b_size(self, val):  # pragma: no cover
+        # Bug in 2024.1, not possible to change cell count.
+        # self._oarray.SetPropValue("B Cell Count", val)
+        pass
+
+    @property
+    def padding_cells(self):
+        """Number of padding cells.
+
+        Returns
+        -------
+        int
+           Number of padding cells.
+        """
+        return int(self._app.get_oo_property_value(self._omodel, self.name, "Padding"))
+
+    @padding_cells.setter
+    def padding_cells(self, val):
+        self._oarray.SetPropValue("Padding", val)
+
+    @property
+    def coordinate_system(self):
+        """Coordinate system.
+
+        Returns
+        -------
+        str
+           Coordinate system name.
+        """
+        cs_dict = self._get_coordinate_system_id()
+        if self._cs_id not in cs_dict.values():
+            self._logger.warning("Coordinate system is not loaded, please save the project.")
```

Comment:
```suggestion
            self._logger.warning("Coordinate system is not loaded. Save the project.")
```
Avoid "please" in technical documentation.